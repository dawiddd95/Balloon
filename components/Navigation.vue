<template>
    <nav class="navigation">
        <div class="navigation__container">
            <NuxtLink to="/">
                <img 
                    class="navigation__logo" 
                    src="~/assets/logo.svg" alt="logo"
                >
            </NuxtLink>
            <NuxtLink class="navigation__cart-button" to="/cart">
                <img class="navigation__cart-icon" src="~/assets/cart.svg" alt="cart icon">
                View Cart
                <!-- Showing product amount in cart if we are in cart page (according to designs) -->
                <div v-if="isCartPage" class="navigation__cart-amount">
                    2
                </div>
            </NuxtLink>
        </div>
    </nav>
</template>
  
<script>

export default {
    name: 'NavigationBar',
    computed: {
        // Check if we are on cart page
        isCartPage() {
            return this.$route.path === '/cart'; // Zakładając, że ścieżka do strony koszyka to '/cart'
        }
    },
};
</script>
  
<style scoped lang="scss">
.navigation {
    width: 100%;
    border-bottom: 1px solid #E9EBEE;

    &__container {
        width: calc(100% - 40px);
        padding: 14px 20px;
        display: flex;
        justify-content: space-between;
        align-items: center;

        @media screen and (min-width: 1160px) {
            width: 1116px;
            margin-left: auto;
            margin-right: auto;
        }
    }

    &__logo {
        cursor: pointer;
    }

    &__cart-button {
        padding: 10px 24px;
        display: flex;
        justify-content: space-between;
        align-items: center;
        color: #201B21;
        font-family: Inter;
        font-size: 13px;
        font-style: normal;
        line-height: 16px;
        border-radius: 8px;
        background-color: white;
        border: 1px solid #201B21;
        cursor: pointer;
        text-decoration: none;

        @media screen and (min-width: 1116px) {
            font-size: 14px;
        }
    }

    &__cart-icon {
        width: 13px;
        height: 16px;
        margin-right: 10px;

        @media screen and (min-width: 1116px) {
            width: 16px;
            height: 24px;
        }
    }

    &__cart-amount {
        width: 16px;
        height: 16px;
        border-radius: 80px;
        background: var(--orange, #EC5E2A);
        color: var(--white, #FFF);
        font-family: Inter Tight;
        font-size: 11px;
        font-weight: 400;
        line-height: 18px;
        display: flex;
        align-items: center;
        justify-content: center;
        margin-left: 8px;
    }
}
</style>