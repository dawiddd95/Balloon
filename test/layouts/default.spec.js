import { shallowMount } from '@vue/test-utils';
import DefaultLayout from '~/layouts/default.vue';
import Navigation from '~/components/Navigation.vue';
import FooterComponent from '~/components/Footer.vue';

describe('Layout default', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallowMount(DefaultLayout, {
            // Stub the <Nuxt /> component since it's a Nuxt-specific component
            stubs: {
                Nuxt: true 
            }
        });
    });

    it('renders correctly', () => {
        expect(wrapper.element).toMatchSnapshot();
    });

    it('contains Navigation component', () => {
        // Checks if the Navigation component is rendered.
        expect(wrapper.findComponent(Navigation).exists()).toBe(true);
    });

    it('contains FooterComponent component', () => {
        // Checks if the FooterComponent component is rendered.
        expect(wrapper.findComponent(FooterComponent).exists()).toBe(true);
    });

    it('renders Nuxt component', () => {
        // Checks if the <Nuxt /> component placeholder is rendered (<nuxt-stub> since Vue Test Utils replaces child components with stubs).
        expect(wrapper.html()).toContain('<nuxt-stub></nuxt-stub>');
    });
});

