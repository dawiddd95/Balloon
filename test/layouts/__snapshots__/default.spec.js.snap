// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout default renders correctly 1`] = `
<div>
  <navigation-stub />
   
  <nuxt-stub />
   
  <footercomponent-stub />
</div>
`;
